tests/unit/lms/resources/_js_config_test.py

```diff
@@ -0,0 +1,161 @@
+import datetime
+from unittest import mock
+
+import jwt
+import pytest
+from pyramid.httpexceptions import HTTPBadRequest
+
+from lms.resources import LTILaunchResource
+from lms.resources._js_config import JSConfig
+from lms.values import HUser
+
+
+class TestJSConfig:
+    """General unit tests for JSConfig."""
+
+    def test_it_is_mutable(self, config):
+        config.update({"a_key": "a_value"})
+
+        assert config["a_key"] == "a_value"
+
+    @pytest.mark.parametrize(
+        "context_property", ["provisioning_enabled", "h_user", "h_groupid"]
+    )
+    def test_it_raises_if_a_context_property_raises(
+        self, context, context_property, pyramid_request
+    ):
+        # Make reading context.<context_property> raise HTTPBadRequest.
+        setattr(
+            type(context),
+            context_property,
+            mock.PropertyMock(side_effect=HTTPBadRequest("example error message")),
+        )
+
+        with pytest.raises(HTTPBadRequest, match="example error message"):
```

Comment:
Generally I think we want to steer clear of checking error text. If `HTTPBadRequest` is something we're worried about the code raising for different reasons we should probably use an error we don't think it can. Maybe `OSError`, or `SyntaxError` or something?